drivers/serial/uart_stm32.c

```diff
@@ -1618,7 +1619,13 @@ static int uart_stm32_init(const struct device *dev)
 		return err;
 	}
 
-	LL_USART_Disable(config->usart);
```

Comment:
This should be kept.
On some series (H7 at least),  device shjould be disabled to set some registers such as baudrate

Reply:
I think RCC reset changes UE bit (Uart Enable) to 0, but I can keep it to be extra safe.

Reply:
ok right. That should be easy to check anyway. But let's keep that for an extra PR if needed.